protocols/relay/src/priv_client.rs

```diff
@@ -219,6 +229,17 @@ impl NetworkBehaviour for Behaviour {
 
         let event = match handler_event {
             handler::Event::ReservationReqAccepted { renewal, limit } => {
+                let addr = self
+                    .direct_connection_addrs
+                    .get(&connection)
+                    .cloned()
+                    .expect("Connection to be direct")
+                    .with(Protocol::P2pCircuit)
+                    .with(Protocol::P2p(self.local_peer_id));
+
+                self.queued_actions
+                    .push_back(ToSwarm::ExternalAddrConfirmed(addr));
```

Comment:
We also need to "expire" the external address once we the reservation expires or our connection to the relay is cut for some reason.

Reply:
We should check `renewal` and _not_ emit a new event if we just renewed a reservation.